src/base58.c

```diff
@@ -44,8 +45,9 @@ int btc_base58_decode(void* bin, size_t* binszp, const char* b58)
     size_t binsz = *binszp;
     const unsigned char* b58u = (const void*)b58;
     unsigned char* binu = bin;
-    size_t outisz = (binsz + 3) / 4;
-    uint32_t outi[outisz];
+    const size_t outisz = (binsz + 3) / 4;
+    const size_t outilen = sizeof(uint32_t) * outisz;
+    uint32_t *outi = (uint32_t *)alloca(outilen);
```

Comment:
Thanks for the explanation. Today I learned!
What makes me a bit worry is the fact that `alloca()` is not a standard C library function while,.. although it looks like that VLAs are "optional" in C++11.

What about trying to get rid of VLAs by just using fixed sizes? Not sure if its possible though.